frappe/public/js/frappe/utils/utils.js

```diff
@@ -1112,75 +1112,104 @@ Object.assign(frappe.utils, {
 		};
 	},
 
-	get_formatted_duration(value, duration_options = null) {
-		let duration = "";
-		if (!duration_options) {
-			duration_options = {
-				hide_days: 0,
-				hide_seconds: 0,
-			};
+	duration_data: {
+		years: { factor: 12 * 1000000000, abbr: "y" },
+		months: { factor: 1000000000, abbr: "mo" },
+		weeks: { factor: 7 * 24 * 60 * 60, abbr: "w" },
+		days: { factor: 24 * 60 * 60, abbr: "d" },
+		hours: { factor: 60 * 60, abbr: "h" },
+		minutes: { factor: 60, abbr: "min" },
+		seconds: { factor: 1, abbr: "s" },
+	},
+
+	/**
+	 * Get formatted duration string.
+	 *
+	 * @param {number} seconds - Duration in seconds
+	 * @param {Object} [duration_options] - Formatting options
+	 * 	Example: { hide_days: true, hide_months: true }
+	 * @returns {string} - Formatted duration string
+	 */
+	get_formatted_duration: function (seconds, duration_options = null) {
+		if (!seconds) {
+			return "";
 		}
-		if (value) {
-			let total_duration = frappe.utils.seconds_to_duration(value, duration_options);
 
-			if (total_duration.days) {
-				duration += total_duration.days + __("d", null, "Days (Field: Duration)");
-			}
-			if (total_duration.hours) {
-				duration += duration.length ? " " : "";
-				duration += total_duration.hours + __("h", null, "Hours (Field: Duration)");
-			}
-			if (total_duration.minutes) {
-				duration += duration.length ? " " : "";
-				duration += total_duration.minutes + __("m", null, "Minutes (Field: Duration)");
+		let components = [];
+		const duration = this.seconds_to_duration(seconds, duration_options);
+		for (const [unit, value] of Object.entries(duration)) {
+			if (value > 0) {
+				components.push(
+					value +
+						__(
+							this.duration_data[unit].abbr,
+							null,
+							unit.toUpperCase() + " (Field: Duration)"
+						)
+				);
 			}
-			if (total_duration.seconds) {
-				duration += duration.length ? " " : "";
-				duration += total_duration.seconds + __("s", null, "Seconds (Field: Duration)");
+		}
+		return components.join(" ");
+	},
```

Comment:
Same, but more generic and using the conversion table.

We’re adding additional units, so this would otherwise have become even longer with even more duplication.